lib/good_migrations/logic.rb

```diff
@@ -1,5 +1,35 @@
 module GoodMigrations
-  class PreventsAppLoad
+  class Logic
+    def self.currently_executing_migration_path
+      migrate_dir_path = Rails.root.join("db/migrate/").to_s
+      loc = caller.detect { |loc| loc.start_with?(migrate_dir_path) }
+      return if loc.nil?
+      loc.partition(":").first
+    end
+
+    def self.extract_time_from_migration_path(path)
+      timestamp_string = File.basename(path).partition("_").first
+      return if timestamp_string.size != 14
```

Comment:
Are you guaranteed that every timestamp string will be 14 characters? Should you not match to initial digits? (e.g. `path.match(/^(\d+)_/)`)?

Reply:
Other than the old migrations which were incremental numbers, i'm not aware of a time where it wasn't a full timestamp. I wouldn't want the string "1500" to be thought of the year 1500.

Since files from which we cannot extract the timestamp will not be permitted to autoload, I thought it would be safer to only accept the known 14 characters ones, and change it system if someone reports a problem.

If you prefer, I could put a minimum of 8 characters, for at least a full date, and let fate decide for the rest.